crates/wasi-preview1-component-adapter/src/lib.rs

```diff
@@ -444,7 +463,7 @@ pub unsafe extern "C" fn fd_allocate(fd: Fd, offset: Filesize, len: Filesize) ->
 /// Note: This is similar to `close` in POSIX.
 #[no_mangle]
 pub unsafe extern "C" fn fd_close(fd: Fd) -> Errno {
-    State::with(|state| {
+    State::with_mut(|state| {
```

Comment:
I had gotten rid of State::with_mut in https://github.com/bytecodealliance/wasmtime/pull/6973, is there a reason this PR brought it back that I haven't spotted, or is this just a git history artifact?

Reply:
`get_input_stream()` now needs to return a `&Descriptor` instead of a `Descriptor` aka `u32`. That lifetime constraint appears to make it incompatible with holding the descriptor table in a `RefCell`. This `State::with_mut` and the current `descriptors(&self) -> &mut Descriptors` code are me searching for a way to describe what this code is doing. I don't think I've found the right answer yet.

Reply:
I've now removed `State::with_mut`.